### PR TITLE
Replace Buzz with HttpClient

### PR DESCRIPTION
**Pull request by renanbr**

### CODE REVIEW

File: src/SymfonyCorp/Connect/Api/Api.php [hunk 1]
```diff
@@ -67,63 +69,87 @@ public function getAccessToken()
...
-        if (400 <= $response->getStatusCode()) {
+            $object = $this->parser->parse($content);
+            $object->setApi($this);
+
+            return $object;
+        } catch (ServerExceptionInterface $exception) {
+            throw new ApiServerException(
+                $response->getStatusCode(false),
+                $response->getContent(false),
+                null,
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/Api/Api.php]: $exception->getMessage()?

File: src/SymfonyCorp/Connect/Api/Api.php [hunk 2]
```diff
@@ -67,63 +69,87 @@ public function getAccessToken()
...
 
-            throw new ApiClientException($response->getStatusCode(), $response->getContent(), $response->getReasonPhrase(), $response->getHeaders(), $error);
-        }
-
-        if (204 === $response->getStatusCode()) {
-            return true;
+            throw new ApiClientException(
+                $response->getStatusCode(false),
+                $response->getContent(false),
+                null,
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/Api/Api.php]: $exception->getMessage()?

File: src/SymfonyCorp/Connect/OAuthConsumer.php [hunk 3]
```diff
@@ -94,39 +95,43 @@ public function requestAccessToken($callbackUri, $authorizationCode)
...
-        $this->logger->debug(sprintf('Response of AccessToken: %s', $response));
+        $this->logger->debug(sprintf('Response of AccessToken: %s', implode("\r\n", [
+            implode("\n", $response->getInfo('response_headers')),
+            $response->getContent(false),
+        ])));
 
-        $content = $response->getContent();
-        $response = json_decode($content, true);
+        try {
+            $responseJson = $response->toArray();
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/OAuthConsumer.php]: let's keep naming it `$response` as way done before

**Comment by renanbr** [comment]
Updates:

- uses `$exception->getMessage()` when possible in cases we wrap exceptions
- keeps `$response` variable naming in `OAuthConsumer`
- branch alias to  `"dev-master": "5.x-dev"`
- bump php dep to `>=7.1.3`
- remove non supported version from .travis.yml

I don't know why travis is not triggered, btw tests are green on local PHP `7.1.28`, `7.2.17` and `7.3.4`

File: composer.json [hunk 4]
```diff
@@ -7,8 +7,10 @@
     "license": "MIT",
     "authors": [],
     "require": {
-        "kriswallsmith/buzz": ">=0.8,<0.17",
-        "psr/log": "^1.0"
+        "psr/log": "^1.0",
+        "symfony/contracts": "1.1.x@dev",
```
Comment (nicolas-grekas) [inline composer.json]: can be removed now that the component is stable

File: composer.json [hunk 5]
```diff
@@ -7,8 +7,10 @@
...
+        "symfony/http-client": "4.3.x@dev",
```
Comment (nicolas-grekas) [inline composer.json]: `^4.3`

File: composer.json [hunk 6]
```diff
@@ -25,7 +27,7 @@
     },
     "extra": {
         "branch-alias": {
-            "dev-migrate-to-symfonyconnect": "5.x-dev"
+            "dev-master": "5.x-dev"
```
Comment (nicolas-grekas) [inline composer.json]: 5.1-dev

File: src/SymfonyCorp/Connect/Api/Api.php [hunk 7]
```diff
@@ -31,15 +33,15 @@ class Api
...
 
-    private $browser;
+    private $httpClient;
     private $parser;
     private $logger;
     private $endpoint;
     private $accessToken;
 
-    public function __construct($endpoint = null, Browser $browser = null, ParserInterface $parser = null, LoggerInterface $logger = null)
+    public function __construct($endpoint = null, HttpClientInterface $httpClient = null, ParserInterface $parser = null, LoggerInterface $logger = null)
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/Api/Api.php]: the type-hint should be replaced by a docblock annotation and passing a `Browser` should trigger a deprecation (and we should then act as if `null` was passed)

File: src/SymfonyCorp/Connect/OAuthConsumer.php [hunk 8]
```diff
@@ -26,22 +27,22 @@ class OAuthConsumer
...
 
-    protected $paths = array(
+    protected $paths = [
         'access_token' => '/oauth/access_token',
         'authorize' => '/oauth/authorize',
-    );
+    ];
 
-    public function __construct($appId, $appSecret, $scope, $endpoint = null, Browser $browser = null, LoggerInterface $logger = null)
+    public function __construct($appId, $appSecret, $scope, $endpoint = null, HttpClientInterface $httpClient = null, LoggerInterface $logger = null)
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/OAuthConsumer.php]: same as Api class

File: src/SymfonyCorp/Connect/OAuthConsumer.php [hunk 9]
```diff
@@ -149,9 +155,9 @@ public function getEndpoint()
         return $this->endpoint;
     }
 
-    public function getBrowser()
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/OAuthConsumer.php]: should be deprecated instead and return a dummy `new Browser(new Curl());`

File: src/SymfonyCorp/Connect/OAuthConsumer.php [hunk 10]
```diff
@@ -149,9 +155,9 @@ public function getEndpoint()
...
+    public function getHttpClient()
```
Comment (nicolas-grekas) [inline src/SymfonyCorp/Connect/OAuthConsumer.php]: if unused, should be removed (and if used, let's double check it's legit and remove if not)

**Comment by nicolas-grekas** [comment]
Continue in #78, thanks @renanbr

**Pull request closed without merging**